tests/dragonfly/replication_test.py

```diff
@@ -1836,3 +1836,46 @@ async def test_replicaof_reject_on_load(df_local_factory, df_seeder_factory):
     await c_replica.close()
     master.stop()
     replica.stop()
+
+
+@pytest.mark.asyncio
+@pytest.mark.slow
+async def test_eviction_propagation(df_local_factory, df_seeder_factory):
+    master = df_local_factory.create(proactor_threads=1, cache_mode="true", maxmemory="290mb")
+    replica = df_local_factory.create(proactor_threads=1)
+    df_local_factory.start_all([master, replica])
+
+    c_master = master.client()
+    c_replica = replica.client()
+
+    seeder = df_seeder_factory.create(port=master.port, val_size=1000, stop_on_failure=False)
+
+    await c_master.execute_command("DEBUG POPULATE 6200000")
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica, timeout=3000)
+
+    # Check data after full sync
+    await check_all_replicas_finished([c_replica], c_master)
+    await check_data(seeder, [replica], [c_replica])
```

Comment:
I belive you can remove this line,
The seeder did not send any data so the keys list the seeder will check is empty

Reply:
yes, thanks.

Reply:
I dont see you removed this line. 
you need to add a check that all the keys in the master are in replica
I can think of few ways to do this but maybe the simplest one whould be scanning master and for each key check that its in replica, and than compare the replica db size and master db size. 
scaning the whole db can take lots of time as you fill so many keys I think we might just compare db size